bool stop

---

string message

bool success
